src/main/resources/webpublic/index.js

```diff
@@ -23,27 +23,51 @@ function showSummary(container, summary) {
 /* Create a box with answer sources. */
 function showSources(container, sources) {
 	container.empty();
-	sources.forEach(function(s) {
+        $.each(sources, function(sid, source) {
 		var state_stags = ['<i>', '<b>', ''];
 		var state_etags = ['</i>', '</b>', ''];
-		container.append('<p class="source">'
-				+ '<img src="/wikipedia-w-logo.png" alt="W" class="wlogo" />'
-				+ ' <a href="http://en.wikipedia.org/?curid='+s.pageId+'" target="_blank">'
-				+ state_stags[s.state] + s.title + state_etags[s.state]
-				+ '</a> (' + s.origin + ')</p>'); // TODO also include the first sentence?
+        if(!(typeof (source.pageId) ==="undefined")) { //this forces to only show en wiki
```

Comment:
Do you mean document title search?  Ahh, you are right.  But I think
page_id not being part of that source is more of a bug (though not
something we need to fix) and we shouldn't rely on it because wa may
want to fix it sometime.

Is there any other way we can solve this?  I think the simplest way
might be to whitelist source types fulltext and title-in-clue, would
that work?  This makes for self-documenting code that clearly spells
out our criteria rather than relying on a bit of an accident with the
pageId.

(Longer term, I expect that we will want to revamp the list of sources
we show anyway - we might want to include some database entries from
Freebase and DBpedia, but we shouldn't be showing too much...  Not sure
how to do it right now.)